### PR TITLE
SingleMuon and Charmonium skim for MUO POG

### PR DESCRIPTION
**Pull request by sscruz**
Configuration for skims of SingleMuon and Charmonium datasets for MUO
 @kfjack  wants to follow this

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@GLP90 
I'm a bit lost in the comments from you that were inlined by github .
The comments were made to a specific commit and it remains unclear which ones remain unresolved.
Please check what's implemented in this PR now and perhaps repeat your comments directly in this PR https://github.com/cms-sw/cmssw/pull/22421/files

File: Configuration/Skimming/python/PDWG_MuonPOGSkim_cff.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+MuonPOGJPsiSkimHLTFilter = copy.deepcopy(hltHighLevel)
+MuonPOGJPsiSkimHLTFilter.throw = cms.bool(False)
+MuonPOGJPsiSkimHLTFilter.HLTPaths = ["HLT_Mu*_Track*_Jpsi*"]
+
+
+TAGMUON_CUT = '(pt > 25) &&  (abs(eta)<2.4) && (isPFMuon>0) && (isGlobalMuon = 1) && (globalTrack().normalizedChi2() < 10) && (globalTrack().hitPattern().numberOfValidMuonHits()>0)&& (numberOfMatchedStations() > 1)&& (innerTrack().hitPattern().numberOfValidPixelHits() > 0)&& (innerTrack().hitPattern().trackerLayersWithMeasurement() > 5) &&  (((pfIsolationR04.sumChargedHadronPt + max(0., pfIsolationR04.sumNeutralHadronEt + pfIsolationR04.sumPhotonEt - 0.5 * pfIsolationR04.sumPUPt) ) / pt)<0.2)'
+TAGMUON_JPSI_CUT = '(isGlobalMuon || numberOfMatchedStations > 1) && pt > 5'
+
+PROBETRACK_CUT = 'pt > 10 &&  abs(eta)<2.4 &&  (charge!=0)'
+PROBETRACK_JPSI_CUT = 'pt > 7 &&  abs(eta)<2.4 &&  (charge!=0)'
```
Comment (GLP90) [inline Configuration/Skimming/python/PDWG_MuonPOGSkim_cff.py]: pt cut should start at 2 GeV for the J/Psi probe

File: Configuration/Skimming/python/PDWG_MuonPOGSkim_cff.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+DIMUON = DIMUON.format(dg0 = "daughter(0).masterClone.standAloneMuon()", dg1="daughter(1)")
+
+
+# Tag and probe for Z#to#mu#mu
+GoodTagMuons = cms.EDFilter("MuonRefPatSelector",
+                            src = cms.InputTag("slimmedMuons"),
+                            cut = cms.string(TAGMUON_CUT)
+                            )
+
+GoodProbeMuons = cms.EDFilter("IsoTrackSelector",
```
Comment (GLP90) [inline Configuration/Skimming/python/PDWG_MuonPOGSkim_cff.py]: To make sure: does this also include Standalone tracks

**Comment by GLP90** [comment]
@slava77 
I have checked the comments again. S

**Comment by GLP90** [comment]
@slava77 
I added the remaining comments on 
https://github.com/cms-sw/cmssw/pull/22421/files
(all the older comments have been fixed by @sscruz)

File: CommonTools/RecoAlgos/plugins/IsoTrackSelector.cc [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "CommonTools/UtilAlgos/interface/SingleObjectSelector.h"
+#include "CommonTools/UtilAlgos/interface/StringCutObjectSelector.h"
+#include "DataFormats/PatCandidates/interface/IsolatedTrack.h"
+
+
+ typedef SingleObjectSelector<
+           pat::IsolatedTrackCollection, 
+           StringCutObjectSelector<pat::IsolatedTrack> 
+         > IsoTrackSelector;
```
Comment (slava77) [inline CommonTools/RecoAlgos/plugins/IsoTrackSelector.cc]: PhysicsTools/PatAlgos/plugins/PATObjectSelector.h seems like a more appropriate place to define this.

File: CommonTools/RecoAlgos/plugins/MuonRefPatSelector.cc [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+#include "CommonTools/UtilAlgos/interface/SingleObjectSelector.h"
+#include "CommonTools/UtilAlgos/interface/StringCutObjectSelector.h"
+#include "DataFormats/PatCandidates/interface/Muon.h"
+
+
+ typedef SingleObjectSelector<
+           pat::MuonCollection, 
+           StringCutObjectSelector<pat::Muon>,
+           pat::MuonRefVector
+         > MuonRefPatSelector;
```
Comment (slava77) [inline CommonTools/RecoAlgos/plugins/MuonRefPatSelector.cc]: PhysicsTools/PatAlgos/plugins/PATObjectSelector.h seems like a more appropriate place to define this.

File: Configuration/Skimming/python/PDWG_MuonPOGSkim_cff.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+                     'keep *_slimmedPatTrigger_*_*',
+                     'keep *_slimmedJets_*_*',
+     )
+)
+
+
+import copy
+from HLTrigger.HLTfilters.hltHighLevel_cfi import *
+from PhysicsTools.PatAlgos.producersLayer1.genericParticleProducer_cfi import patGenericParticles
+from PhysicsTools.PatAlgos.producersLayer1.muonProducer_cfi import patMuons
```
Comment (slava77) [inline Configuration/Skimming/python/PDWG_MuonPOGSkim_cff.py]: why are the patGenericParticles and patMuons needed here?
please remove if not required

File: Configuration/Skimming/python/autoSkim.py [hunk 6]
```diff
@@ -32,6 +34,8 @@
     'Mu' : 'WMu+ZMu+HighMET+LogError',    
     'EG':'WElectron+ZElectron+HighMET+LogError',
     'TopMuEG':'TopMuEG+LogError',
+    'MuonPOG': 'MuonPOGSkim+LogError',
+    'MuonJPsiPOG': 'MuonPOGJPsiSkim+LogError',
```
Comment (slava77) [inline Configuration/Skimming/python/autoSkim.py]: I'm curious why these are only in ```autoSkimRunI ``` 
I'm not well aware of the use cases though.

Comment (kfjack) [reply Configuration/Skimming/python/autoSkim.py]: I think these two lines should be removed. Also we do not have a mapping of MuonPOG/MuonJPsiPOG PD.

Comment (sscruz) [reply Configuration/Skimming/python/autoSkim.py]: Hi, I've removed those lines from autoSkim. @kfjack can you point me have to do the mapping?

Comment (kfjack) [reply Configuration/Skimming/python/autoSkim.py]: I assume MuonPOGSkim skim should run on single Mu PD, and MuonPOGJPsiSkim should run on Charmonium PD, right? You can just add the skim to 
'SingleMuon' : 'ZMu+MuTau+LogError+LogErrorMonitor',
'Charmonium' : 'BPHSkim+LogError+LogErrorMonitor',
e.g. 
'SingleMuon' : 'MuonPOGSkim+ZMu+MuTau+LogError+LogErrorMonitor',
'Charmonium' : 'MuonPOGJPsiSkim+BPHSkim+LogError+LogErrorMonitor',

**Comment by perrotta** [comment]
please test

**Comment by slava77** [comment]
the error is related.
Some edits are missing in PhysicsTools/PatAlgos/plugins/PATObjectSelector.cc to match what was done in PATObjectSelector.h

**Comment by perrotta** [comment]
please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/22421	2161faa
- relatively simple changes in what regards reco signature: IsoTrackSelector and MuonRefPatCount selectors/filters are added for pat objects in PhysicsTools/PatAlgos/plugins/PATObjectSelector
- jenkins tests pass

**Comment by fabozzi** [comment]
@kfjack These skims should be integrated in data relvals for regular testing (this can be done in a separate PR)

**Comment by fabozzi** [comment]
+1

**Comment by kfjack** [comment]
@fabozzi OK will take care of the relval afterwards!

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

**Pull request merged**